sqld/src/lib.rs

```diff
@@ -500,6 +502,24 @@ async fn run_storage_monitor(db_path: PathBuf, stats: Stats) -> anyhow::Result<(
     Ok(())
 }
 
+fn sentinel_file_path(path: &Path) -> PathBuf {
+    path.join(".sentinel")
+}
+/// initialize the sentinel file. This file is created at the beginning of the process, and is
+/// deleted at the end, on a clean exit. If the file is present when we start the process, this
+/// means that the database was not shutdown properly, and might need repair. This function return
+/// `true` if the database is dirty and needs repair.
+fn init_sentinel_file(path: &Path) -> anyhow::Result<bool> {
```

Comment:
I find it the return value a little counter-intuitive, because the `init_sentinel_file()` name doesn't suggest that it returns a decision if the database needs repair, and at call sites, I'd expect that when `init_something` returned true, it just succeeded. But the comment is there to explain the semantics and the call site named the variable `db_is_dirty`, so it's clear enough, lgtm